src/lib/Parser.hs

```diff
@@ -365,10 +385,10 @@ oneLabeledItem bindwith itemparser punner = do
     Nothing -> explicitBound
   return $ labeledSingleton l itemVal
 
-interfaceRecordFields :: String -> Parser (ExtLabeledItems UExpr UExpr)
+interfaceRecordFields :: String -> Parser (LabeledItems UExpr)
 interfaceRecordFields bindwith =
   fuse <$> onePerLine (oneLabeledItem bindwith expr (Just varPun))
```

Comment:
Passing `Just varPun` here means that you are allowing variable punning. Thus
```
interface ... where
  foo
  bar
```
will desugar to
```
interface ... where
  foo : foo
  bar : bar
```
This seems pretty nonsensical to me. (See above my comment about `oneLabeledItem`; I think it would be better to just inline the parts you want instead of trying to abstract out a helper function)